clouditor-engine-aws/src/main/java/io/clouditor/discovery/aws/AwsS3BucketScanner.java

```diff
@@ -86,17 +86,28 @@ public Asset transform(Bucket bucket) throws ScanException {
         // needed, as long as https://github.com/aws/aws-sdk-java-v2/issues/52 is not fixed
         LOGGER.info("Switching to region-specific S3 client ({})", region);
 
-        var account = PersistenceManager.getInstance().getById(AwsAccount.class, "AWS");
-
-        if (account == null) {
-          throw SdkClientException.create("AWS account not configured");
-        }
+        var account =
+            new HibernatePersistence()
```

Comment:
More from a Codestyle perspective, it seems that every time a DB call is made, a new `HibernatePersistence` object is initiated. Is that really necessary? Do we have access to the `Engine`  object in the scanners and maybe have a common persistence object there? Any specific reason, not to use the existing singleton-pattern?

Reply:
We don´t need one global instance of a Persistence manager anymore. So it is possible to avoid the violence of the Single-responsibility principle.